scipy/fftpack/czt.py

```diff
@@ -0,0 +1,535 @@
+
+
+def czt(x, m=None, w=1.0, a=1, axis=-1):
+    """
+    Compute the frequency response around a spiral.
+
+    Parameters:
+    ----------
+    x : array
+        The set of data to transform.
+    m : int, optional
+        The number of points desired. Default is the length of the input data.
+    w : complex or float, optional
+        If w is complex, it is the ratio between points in each step.
+        If w is float, it is the frequency step scale (relative to the
+        normal dft frequency step).
+    a : complex, optional
+        The starting point in the complex plane.  Default is 1.
+    axis : int, optional
+        Array dimension to operate over.  Default is the final dimension.
+
+    Returns:
+    -------
+    An array of the same dimensions as x, but with the length of the
+    transformed axis set to m.  Note that this is a view on a much
+    larger array.  To save space, you may want to call it as
+    y = ascontiguousarray(czt(x))
+
+    See zoomfft for a friendlier interface to partial FFT calculations.
+
+    If the transform needs to be repeated, use CZT to construct a
+    specialized transform function which can be reused without
+    recomputing constants.
+    """
+    x = np.asarray(x)
+    transform = CZT(x.shape[axis], m=m, w=w, a=a)
+    return transform(x, axis=axis)
+
+
+def zoomfft(x, fn, m=None, Fs=2, axis=-1):
+    """
+    Compute the Fourier transform of x for frequencies in [f1, f2].
+
+    Parameters:
+    ----------
+    x : array
+        The input signal.
+    fn : array_like, optional
+        A scalar or length-2 sequence giving the frequency range. If a scalar,
+        the range 0-fn is assumed.
+    m : int, optional
+        The number of points to evaluate.  The default is the length of x.
+    Fs : float, optional
+        The sampling frequency.  With a sampling frequency of
+        10kHz for example, the range f1 and f2 can be expressed in kHz.
+        The default sampling frequency is 2, so f1 and f2 should be
+        in the range 0,1 to keep the transform below the Nyquist
+        frequency.
+    axis : int, optional
+        The array dimension the transform operates over.  The default is the
+        final dimension.
+
+    Returns:
+    -------
+    array
+        The transformed signal.  The Fourier transform will be calculate
+        at the points f1, f1+df, f1+2df, ..., f2, where df=(f2-f1)/m.
+
+    Notes
+    -----
+    ``zoomfft(x, [0, 2-2./len(x)])`` is equivalent to ``fft(x)``.
+
+    To graph the magnitude of the resulting transform, use::
+
+        plot(linspace(f1, f2, m), abs(zoomfft(x, [f1, f2], m)))
+
+    If the transform needs to be repeated, use ZoomFFT to construct
+    a specialized transform function which can be reused without
+    recomputing constants.
+    """
+    x = np.asarray(x)
+    transform = ZoomFFT(x.shape[axis], fn, m=m, Fs=Fs)
+    return transform(x, axis=axis)
+
+
+def _test1(x, show=False, plots=[1, 2, 3, 4]):
```

Comment:
If a test, belongs in the test file. If intended more as a demo/debugging tool, might include it in the documentation somewhere.